reana/cli.py

```diff
@@ -1695,6 +1720,61 @@ def python_install_eggs():
             run_command(cmd, component)
 
 
+@cli.command(name='python-unit-tests')
+@click.option('--component', '-c', multiple=True, default=['ALL'],
+              help='Which components? [shortname|name|.|CLUSTER|ALL]')
+def python_unit_tests(component):  # noqa: D301
+    """Run Python unit tests in independent environments.
+
+    For each component, create a dedicated throw-away virtual environment,
+    install latest shared modules (reana-commons, reana-db) that are currently
+    checked-out and run the usual component unit tests. Delete the throw-away
+    virtual environment afterwards.
+
+    \b
+    :param components: The option ``component`` can be repeated. The value may
+                       consist of:
+                         * (1) standard component name such as
+                               'reana-workflow-controller';
+                         * (2) short component name such as 'r-w-controller';
+                         * (3) special value '.' indicating component of the
+                               current working directory;
+                         * (4) special value 'CLUSTER' that will expand to
+                               cover all REANA cluster components [default];
+                         * (5) special value 'CLIENT' that will expand to
+                               cover all REANA client components;
+                         * (6) special value 'DEMO' that will expand
+                               to include several runable REANA demo examples;
+                         * (7) special value 'ALL' that will expand to include
+                               all REANA repositories.
+    :type component: str
+    """
+    components = select_components(component)
+    for component in components:
+        if is_component_python_package(component):
+            cmd_activate_venv = \
+                'source  ~/.virtualenvs/_{}/bin/activate'.format(component)
+            for cmd in ['virtualenv ~/.virtualenvs/_{}'.format(component),
+                        '{} && which python'.format(cmd_activate_venv),
+                        '{} && cd ../pytest-reana && '
+                        ' pip install . --upgrade'.format(cmd_activate_venv),
+                        '{} && cd ../reana-commons && '
+                        ' pip install . --upgrade'.format(cmd_activate_venv),
+                        '{} && cd ../reana-db && '
+                        ' pip install . --upgrade'.format(cmd_activate_venv),
+                        'git clean -d -ff -x',
+                        '{} && pip install ".[tests]" --upgrade'.format(
+                            cmd_activate_venv),
+                        '{} && python setup.py test'.format(cmd_activate_venv),
```

Comment:
Why we don't call `./run-tests.sh` here? In https://github.com/reanahub/reana-job-controller/pull/244#issuecomment-608346000 there is work related to moving the invocation of tests inside the RJC image because we cannot run RJC tests locally on Mac (no HTCondor Python package on Mac).

Reply:
Ouch, macOS.

I did not want to run `run-tests.sh` because of speed concerns. E.g. we are running `docker build...` inside `run-tests.sh`, which would considerably lengthen the execution, and could cause either mixes with running REANA system or else would eat another disk space.  E.g. checking Sphinx docs, manifest etc is good, but we can perhaps leave this onto Travis CI build?

The command was conceived as a quick "run reasonable Python tests quickly" locally without  messing much with my REANA instance... That kind of feeling.

We can chat IRL whether that's the most common use case for you as well?  We could introduce a flag to distinguish the two perhaps.

Reply:
I was thinking of the problem regarding the long execution time because of the builds, and all solutions are just adding complexity. Therefore, being pragmatic, I would be in favor of a flag to exclude components, so on macOS we could run everything except RJC (even though excluding code from a test command sounds wrong, but [even pytest does it](http://doc.pytest.org/en/latest/skipping.html)).

Reply:
Another option, if you can *never* run `r-j-controller`'s unit tests on macOS, would be to simply detect the underlying OS and to skip the `r-j-controller` tests and emit a warning to user that either this whole component is skipped or that at least some tests are skipped (once we patch `r-j-controller` tests to separate htc ones from non-htc ones, so that at least *some* unit tests would run on macOS. In this way people won't even have to type `--exclude-components`. WDYT?

Reply:
Added detection of `macOS`.